media/js/firefox/sticky-promo.js

```diff
@@ -43,20 +53,40 @@
         };
 
         StickyPromo.show = function (){
-        // Open promo
-            Mzp.StickyPromo.open();
+            var stickyPromoShowInit = false;
+            document.addEventListener('scroll', function () {
+                if (!stickyPromoShowInit) {
+                    // Open promo
+                    Mzp.StickyPromo.open();
+                    stickyPromoShowInit = true;
+                }
+            });
 
             // Add modifier class to the footer to make sure the language selection drop-down is not obscured by the sticky promo
             var footer = document.querySelector('.c-footer');
             if (footer) {
                 footer.classList.add('is-intersecting-sticky-overlay');
+
+                document.addEventListener('scroll', function () {
```

Comment:
This event fires a lot to check for the footer. Might it be possible to throttle / debounce the handler so it only needs to check every 100ms or so? Alternatively, it might be worth considering [Intersection Observer](https://developer.mozilla.org/en-US/docs/Web/API/Intersection_Observer_API) perhaps if performance is a concern?

Reply:
@alexgibson Great call out. Looking at the docs for  Intersection Observer, it looks like it is lacking any IE support. Should that be avoided?

Reply:
@maxxcrawford I'll defer to you on what you think is ultimately the most appropriate thing to use here. But as long as we use feature detection to avoid errors on older browsers, then we should be ok. A couple of options might be:

a) Make Intersection Observer a requirement for user agents that see the sticky promo (IE users wouldn't get it).
b) Fallback to a scroll event handler for browsers that don't support the newer method (wider feature coverage, but more complex).

Personally I think a) might be ok here imho. All modern browsers support Intersection Observer today, so as long as we consider the promo like a form of progressive enhancement, then IE visitors just get the regular page and be none the wiser.

Reply:
Debounce added! I used [the function](https://davidwalsh.name/javascript-debounce-function) from @darkwing. 🙌